### PR TITLE
WIP: start detecting react synthetic events

### LINKED ISSUE #600: withReact incorrectly generates dom event names
withReact() incorrectly assumes that **react's** event names* (passed down as component's props) can be converted to DOM events names using dasherize() fn. While this works for names like 'click', it fails on others,  for example: 

- onDoubleClick => double_click, while **dblclick** is needed  (this one is probably the only one exception in naming according to react-dom source code)
- onMouseOver => mouse_over, while **mouseover** is needed

```
js/with-react.js on line 115:
const eventName = dasherize(key.substring(2));
```

https://reactjs.org/docs/events.html
https://www.w3schools.com/jsref/dom_obj_event.asp

## Are you reporting a bug or a feature?

 - [x ] Bug
 - [ ] Feature

## Expected Behavior

<AXAButton onDoubleClick={()=>alert('double')}> works

## Actual Behavior

<AXAButton onDoubleClick={()=>alert('double')}> does nothing
<AXAButton onDblclick={()=>alert('double')}> does work instead

### PR DESCRIPTION
**Pull request by AndyOGo**
Fixes #600 

Changes proposed in this pull request:

 - Adds distinction of namespaced custom events and React's synthetic events
 - How to continue???

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
 
 # How Has This Been Tested?
 
 
 
 # Checklist:
 
 - [ ] My code follows the style guidelines of this project
 - [ ] I have performed a self-review of my own code
 - [ ] I have commented my code, particularly in hard-to-understand areas
 - [ ] I have made corresponding changes to the documentation
 - [ ] My changes generate no new warnings
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/js/with-react.js [hunk 1]
```diff
@@ -108,6 +112,14 @@ const withReact = (WebComponent, { pure = true, passive = false } = {}) => {
       const [eventKeys, dataKeys] = propsKeys.reduce(partition(isEventFilter), [[], []]);
 
       eventKeys.forEach((key) => {
+        // supper React's synthetic events
+        // Note: not 100% compatible with custom elements
+        if (!isNamespacedEvent(key)) {
```
Comment (AndyOGo) [inline src/js/with-react.js]: @MarekLacoAXA 
@LucaMele 
To support React's synthetic events we would need to implement some special handling:
https://reactjs.org/docs/web-components.html

**Comment by MarekLacoAXA** [comment]
Do I understand right - does this change mean, that any non-axa-prefixed event names (onXxx attributes comming from react) will now get ignored?

Just curious - where in the PL's src can I find the code:
- where the handleClick() in button.js gets called
- where the axa_click event gets invoked

Thanks for the links in the bugreport!

**Comment by AndyOGo** [comment]
**Breaking Changes:** Please migrate before merging
https://github.com/axa-ch/patterns-library/releases/tag/v2.0.1-beta.217

**Comment by AndyOGo** [comment]
@LucaMele 

I would like to get this merged.

Basically it hides all custom `onaxa*` events from React and defers handling of those to the custom element itself.
All normal react events will be attached as normal.

**Comment by LucaMele** [comment]
@AndyOGo has this breaking changes? ergo for all those who use with-react so far, do they have to change name of the callback function? Ergo `<AnyReactComp onAxaChange={()=>{}}` to `<AnyReactComp onChange={()=>{}}` or is it backwards compatible?

**Comment by AndyOGo** [comment]
@LucaMele 
There should not be any breaking changes.
The idea is to make sure, that:
- all custom namespaced `onaxa*` events are being forward to the custom element only
- all react's synthetic `on*` events never touch the custom element and stay in the react world

**Comment by AndyOGo** [comment]
Tested locally, though I can't see in the tty when the build is actually finished 🤔

**Pull request merged**